### ISSUE TITLE
Hana: create procedure fails with syntax error at END if there is a loop inside.

### ISSUE THREAD

**Issue by wernerdaehn** [dbeaver/dbeaver#13745]
#### System information: 
- Win10
- DBeaver 21.2.0

#### Connection specification:
Sap Hana

#### Describe the problem you're observing:
Simple procedure cannot be created due to supposed syntax errors.

#### Steps to reproduce, if exist:
Run the SQL Script:

```
create or replace procedure abc
      (in x SMALLINT,
       in y SMALLINT)      
as 
begin
    declare loop_x SMALLINT;
    declare loop_y SMALLINT;
    
    declare maxID BIGINT;
    
    for loop_y in 1..:y do
        for loop_x in 1..:x do
            maxID := maxID + 2;
        end for;
    end for;    
end;
```

It seems the parser sends only the string
```
create or replace procedure abc
      (in x SMALLINT,
       in y SMALLINT)      
as 
begin
    declare loop_x SMALLINT;
    declare loop_y SMALLINT;
    
    declare maxID BIGINT;
    
    for loop_y in 1..:y do
        for loop_x in 1..:x do
            maxID := maxID + 2;
        end for
```
truncating the statement after the end-for's semicolon char.

Error message is
```
SQL Error [257] [HY000]: SAP DBTech JDBC: [257] (at 302): sql syntax error: line 14 col 13 (at pos 302)
```

and the create statement is obviously valid, it works in Hana Studio.

**Comment by LonwoLonwo** [comment]
Hello @wernerdaehn 

Thanks for the bug report!

You can try to highlight your procedure statement in SQL Editor and use Ctrl+Enter to avoid block parsing problems.

Now we, unfortunately, cannot test this case, because the connection to the database you gave us is no longer working. If you have the opportunity to back our access to the database - it would be very cool!

**Comment by wernerdaehn** [comment]
I have shutdown the server but can start it at any time. It is just that I do no longer need it 24x7. Please send me an email when and for how long I should start the server and I will do so.

**Comment by LonwoLonwo** [comment]
Thanks for your help. Your database was very useful for us.

Anyway. I see a blank line in your statement.
Can you please tell me - the "Blank line is statement delimiter" is enabled or disabled in your preferences?

![2021-09-01 18_13_04-DBeaver Ultimate 21 2 0 - _orcl_ Script-8](https://user-images.githubusercontent.com/45152336/131697461-af599285-0a56-428b-b6f6-5d40331425f2.png)

**Comment by wernerdaehn** [comment]
Yes, it was checked. It tried with and without (opening a new editor after the change to be safe).
I also removed the blank lines from the procedure code.
Behavior is always the same, the code sent to the database ends at the line `end for;`

**Comment by LonwoLonwo** [comment]
Fixed in DBeaver 21.2.2

**Comment by DariaMarkaryan** [comment]
Veified
